### DISCUSSION TITLE
How achieve Media Library Clone or segregation of Media Libraray

### DISCUSSION THREAD

**Discussion by saleemhaq** [OrchardCMS/OrchardCore#11429] [Q&A]
Please help me on below:

![teser](https://user-images.githubusercontent.com/5202291/159736858-a3f2b819-02d1-4b2a-9580-5e3ca08d0775.png)

How i achieve I suffer from last week to till now folks please help.

**Comment by jptissot** [comment]
@Skrypt Did we ever implement folder permissions on the Media folders ?

**Reply by saleemhaq** [reply]
> @Skrypt Did we ever implement folder permissions on the Media folders ?

@jptissot can you help me out on above concern?

**Reply by Skrypt** [reply]
No, we did not, but there's surely a quick way to implement a middleware that will do. I remember vaguely having a discussion about this with @deanmarcussen

**Reply by saleemhaq** [reply]
> No, we did not, but there's surely a quick way to implement a middleware that will do. I remember vaguely having a discussion about this with @deanmarcussen
@Skrypt can you guide me about the way out of above requirement?

**Reply by Skrypt** [reply]
See https://github.com/OrchardCMS/OrchardCore/issues/9369 as issue reference. Nothing done yet.

**Comment by jptissot** [comment]
@saleemhaq Can't you just create a hierarchy of folders ? Would that work? Or do you need a completely separate media library for both purposes.

**Reply by saleemhaq** [reply]
> @saleemhaq Can't you just create a hierarchy of folders ? Would that work? Or do you need a completely separate media library for both purposes.

Need completely separate media library? @jptissot

**Comment by saleemhaq** [comment]
Need completely separate media library?

**Reply by jptissot** [reply]
I suggest you look into implementing some type of permission system on the Media folders and that way a user will only see what they need to see. It's hard to advise you on the best way to go since I don't know your requirements.

**Comment by saleemhaq** [comment]
Just need to give another media library option like media library 2 and it
has the same all functionality like Media Library have.

**Reply by sebastienros** [reply]
You can create your own module with a different name and folder to store them.

**Comment by saleemhaq** [comment]
How i clone the existing one end to end? please advise. so will it act the
media library 1, thanks in advance.

**Reply by saleemhaq** [reply]
> How i clone the existing one end to end? please advise. so will it act the media library 1, thanks in advance.

Folks please advise me on below please:
How i clone the existing one end to end? please advise. so will it act the
media library 1, thanks in advance. @sebastienros

**Reply by ns8482e** [reply]
Source is open, look at media module and make customization as you need

**Reply by Skrypt** [reply]
Copy the module. Rename it, rename all the assemblies and customize for your need.

**Reply by hishamco** [reply]
As the guys said, the source is open, so fork the repo and do your customization then
